### DISCUSSION TITLE
Memory leak prevention

### DISCUSSION THREAD

**Discussion by wizzymore** [alpinejs/alpine#4025] [1. Help] [answered]
I have a lot of objects that are initialized on the window and needs to be destroyed, they are created in Alpinejs and are custom libraries like charts, datatables etc, etc...

When navigating with wire:navigate, how do i know when the component itself is unmounted so i can release the custom events and unregister the objects?
This is causing a lot of memory leaks and i can't seem to find an issue. I header about an undocumented "destroy" directive but it doesn't work for me, i tried with x-destroy, `destroy() {}` on the data object, nothing works. I also looked in the source code and can't find a directive like this.

From what it looks like, there is already a cleanup function that alpinejs calls internally, why can't we access it?

**Comment by ekwoka** [comment, marked as answer]
`destroy` method on the component does work. What makes you think it doesn't?

Also, all directives unregister themselves.

the cleanup is used inside directives and such.

For your components, you should use `destroy`

**Reply by wizzymore** [reply]
Yes, i figured it out in the end, there was something wrong with the dependencies of the project. Fixing this made alpine work ok again and destroy is now called correctly.

Thank you for your time.

**Reply by ekwoka** [reply]
Depending on the nature of the things, a custom directive could be a decent way to handle this. You get access to more of those internal lifecycles that way.
